project-addons/flask_middleware_connector/models/stock.py

```diff
@@ -32,10 +35,34 @@ class StockMove(models.Model):
 
     @api.multi
     def write(self, vals):
+        """
+        Incluir la comprobacion de todas las lineas del albaran despues de esta funcion.
+        Si un albaran tiene alguna linea esperando disponibilidad, su estado sera parcialmente disponible.
+        Si un albaran tiene todas las lineas listas, ninguna esperando disponibilidad y ninguna o \
+        varias lineas canceladas, su estado tiene que ser listo para transeferir. 
+        SI un albaran tiene todas sus lineas canceladas su estado pasa a cancelado. 
+        Si un albaran tiene todas sus lineas esperando disponibilidad y ninguna\
+        o alguna linea cancelada, su estado sera esperando disponibilidad  
+        """
+        old_state = False
+        new_state = False
+        if self:
```

Comment:
sobra el if, siempre habrá self

Reply:
De todas formas, no se si estos estara bien, en el caso de que en self vinieran más de un id....

Reply:
Si, siempre hay self, pero no siempre tiene un objeto dentro. A veces intentaba coger el estado del self y me daba un fallo porque self contenía unicamente "stock.move()"

Reply:
Todos los move que vengan en self deberían de venir relacionados sobre un mismo albarán ¿No?

Reply:
al write debe entrar con el self rellenado en la nueva api, sino no va a escribir nada, no debería de funcionar, de todas formas yo creo que necesitas meterlo dentro de un "for" por lo que esto ya no sería necesario, ya que sino viniera nada ya no entraría en el for, aunque ya te digo que no debería de pasar.

Un move se puede escribir en algún caso antes de asociarlo a un albarán. Los abstecimientos crean movimientos y al ejecutarse se crean albaranes, peor hasta que se ejecuten son simples movimientos sin albarán